### DISCUSSION TITLE
Ability to register packages via DI container

### DISCUSSION THREAD

**Discussion by Wulfheart** [dedoc/scramble#199] [Q&A]
Related to https://github.com/dedoc/scramble/pull/189

Hello,

this use case might be a little bit special but I am going to propose it nevertheless.
I am trying to register extensions via the DI container as this would allow me to create a base package with all my custom extensions. What do you think about it?

**Comment by Wulfheart** [comment]
My current solution is this in the package but feels hacky af.

```php
        if (! ($this->app instanceof CachesConfiguration && $this->app->configurationIsCached())) {
            /** @var Repository $config */
            $config = $this->app->make('config');

            $alreadyRegisteredExtensions = $config->get('scramble.extensions', []);
            $config->set('scramble.extensions', [
                ...$alreadyRegisteredExtensions,
                RuleToSchemaExtension::class
            ]);
        }
```

**Reply by romalytvynenko** [reply]
Oooh, that's interesting!

LOL, I always hoped people will create packages for other libs for Scramble, but never thought about API for this 👯 

Thank you so much for reporting, I will definitely do something about it.

I'm going to check PHPStan/Psalm and other and will provide similar API for this issue.

Thanks for sharing 🫶
